agent/doc/swdesign/README.md

```diff
@@ -853,7 +944,383 @@ Needs:
+and the workload is an inter-workload dependency of at least one other workload
+and the workload has at least one unfulfilled `DeleteCondition`,
+the WorkloadScheduler shall report the workload state `Stopping(WaitingToStop)`.
+
+Rationale: This communicates to the user that the delete of the inter-workload dependency is scheduled but not ready because another workload requires it.
+
+Tags:
+- WorkloadScheduler
+
+Needs:
+- impl
+- utest
+- stest
+
+#### The agent reports workload state stopping for updated workloads with unfulfilled DeleteConditions
+`swdd~agent-reports-pending-delete-workload-state-on-pending-update-delete~1`
+
+Status: approved
+
+When the WorkloadScheduler receives the `WorkloadOperation` to update a workload
+and the workload is an inter-workload dependency of at least one other workload
+and the workload has at least one unfulfilled `DeleteCondition`,
+then the WorkloadScheduler shall report the workload state `Stopping(WaitingToStop)`.
+
+Rationale: This communicates to the user that the delete of the inter-workload dependency is scheduled but not ready because another workload requires it.
+
+Tags:
+- WorkloadScheduler
+
+Needs:
+- impl
+- utest
+- stest
+
+#### The agent reports workload state pending for updated workloads with fulfilled DeleteConditions
+`swdd~agent-reports-pending-create-workload-state-on-pending-update-create~1`
+
+Status: approved
+
+When the WorkloadScheduler receives the `WorkloadOperation` to update a workload
+and the workload is an inter-workload dependency of at least one other workload
+and the workload has fulfilled `DeleteCondition`s,
+then the WorkloadScheduler shall report the workload state `Pending(WaitingToStart)`.
+
+Rationale: This communicates to the user that the create operation of that workload is scheduled but not ready because the create operation of the update waits for inter-workload dependencies and the create operation shall only run after completing the delete operation of the update with the default update strategy `AT_MOST_ONCE`.
+
+Tags:
+- WorkloadScheduler
+
+Needs:
+- impl
+- utest
+- stest
+
+#### The agent enqueues a delete only operation of an update
+`swdd~agent-shall-not-enqueue-update-delete-only-workload-operation~1`
+
+Status: approved
+
+The WorkloadScheduler shall not put the update delete only `WorkloadOperation` into a waiting queue.
+
+Rationale: The update delete only workload operation is internally created when the delete operation of a pending update is ready.
+
+Tags:
+- WorkloadScheduler
+
+Needs:
+- impl
+- utest
+
+#### A workload is ready to create when all of its inter-workload dependencies are fulfilled
+`swdd~workload-ready-to-create-on-fulfilled-dependencies~1`
+
+Status: approved
+
+When the workload state of all inter-workload dependencies of a workload fulfill their configured `AddCondition`, then the Workload is ready to create.
+
+Tags:
+- WorkloadScheduler
```

Comment:
We are not mentioning the WorkloadScheduler in this requirement at all and I don't think we should tag it here. I understand that you want to include the DependencyStateValidator in the chain, but maybe this requirement here is only for it and the DependencyStateValidator should be mentioned above in the requirements that check the "readiness".